xdsl/dialects/riscv_scf.py

```diff
@@ -74,10 +74,63 @@ def __init__(
         )
 
 
+@irdl_op_definition
+class WhileOp(IRDLOperation):
+    name = "riscv_scf.while"
+    arguments: VarOperand = var_operand_def(RISCVRegisterType)
+
+    res: VarOpResult = var_result_def(RISCVRegisterType)
+    before_region: Region = region_def()
+    after_region: Region = region_def()
+
+    def __init__(
+        self,
+        arguments: Sequence[SSAValue | Operation],
+        result_types: Sequence[IntRegisterType],
+        before_region: Region | Sequence[Operation] | Sequence[Block],
+        after_region: Region | Sequence[Operation] | Sequence[Block],
+    ):
+        super().__init__(
+            operands=[arguments],
+            result_types=[result_types],
+            regions=[before_region, after_region],
+        )
+
+    # TODO verify dependencies between riscv_scf.condition, riscv_scf.yield and the regions
+    def verify_(self):
+        for idx, arg in enumerate(self.arguments):
+            if self.before_region.block.args[idx].type != arg.type:
```

Comment:
Can we use `zip` instead of `enumerate` + `idx`?

Reply:
We could, but I prefer this version because when I break on a line like this with an exception breakpoint I get the index directly.

Reply:
Addressed in newest commit. Good to go.